cmd/device-hub-cli/helpers.go

```diff
@@ -162,33 +221,31 @@ func roundTrip(sample interface{}, fn roundTripFunc) error {
 			fmt.Println(fi.Name())
 
 			folderPath := path.Join(cfg.RequestDir, fi.Name())
-			f, d, err := decoderFromPath(folderPath)
 
+			var data cliConf
+			err = data.Load(folderPath)
 			if err != nil {
 				return err
 			}
-
-			decoders = append(decoders, d)
-			files = append(files, f)
-
+			dataSlice.Append(data)
 		}
 	}
 
-	defer func() {
-		for i, _ := range files {
-			files[i].Close()
-		}
-	}()
-
 	conn, client, err := dial()
 	if err != nil {
 		return err
 	}
-
 	defer conn.Close()
+	// sort the config items to create & delete cases
+	switch caller {
+	case "create":
+		sort.Sort(cliConfSlice(dataSlice))
+	case "delete":
+		sort.Sort(sort.Reverse(cliConfSlice(dataSlice)))
+	}
```

Comment:
I'm not 100 sure that this switch on the caller is required. I understand what you are trying to do - its just leading to a bit of spaghetti code. Maybe the ordering could be expressed clearer before the roundtrip function?